Bottom: 726bcb92d78f82893fbcfb516d2f8f5a30411a96
Top:    19b7cd20e9a755de1e4bdae35871abe2daa4650d
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-09-12 01:22:10 +0100

Debug / attempt to fix DRC crash


---

diff --git a/src/find.c b/src/find.c
index efd3ebf..b84c8b7 100644
--- a/src/find.c
+++ b/src/find.c
@@ -2680,6 +2680,10 @@ IsLineInPolygon (LineTypePtr Line, PolygonTypePtr Polygon)
        BDimension wid = (Line->Thickness + Bloat + 1) / 2;
        LocationType x1, x2, y1, y2;
 
+       printf ("Line thickness is %i, Bloat is %i\n",
+               Line->Thickness, Bloat);
+       printf ("Testing with wid=%i\n", wid);
+
        x1 = MIN (Line->Point1.X, Line->Point2.X) - wid;
        y1 = MIN (Line->Point1.Y, Line->Point2.Y) - wid;
        x2 = MAX (Line->Point1.X, Line->Point2.X) + wid;
diff --git a/src/polygon.c b/src/polygon.c
index 3e7b3e5..3b302c3 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -1683,6 +1683,7 @@ IsRectangleInPolygon (LocationType X1, LocationType Y1, LocationType X2,
                       LocationType Y2, PolygonTypePtr p)
 {
   POLYAREA *s;
+  printf ("X1=%i, X2=%i, Y1=%i, Y2=%i\n", X1, X2, Y1, Y2);
   if (!
       (s = RectPoly (min (X1, X2), max (X1, X2), min (Y1, Y2), max (Y1, Y2))))
     return false;
